src/content/developers/tutorials/integrating-your-smart-contract-with-the-frontend/index.md

```diff
@@ -0,0 +1,920 @@
+
+And if you haven't already, you'll definitely need an Alchemy account to complete any of these tutorials. Sign up for a free account [here](https://alchemy.com/?a=eth-org-frontend-integration)!
+
+## Part 4: Marrying Web2 & Web3: connecting your smart contract to a frontend project
+
+🎉Woooo! You finally made it to the last part of this tutorial series: creating a full stack decentralized application \(dApp\) by connecting your Hello World smart contract to a frontend project and interacting with it.
+
+By the end of this tutorial, you'll know how to:
+
+* Connect a Metamask wallet to your dApp project
+* Read data from your smart contract using the [Alchemy Web3](https://docs.alchemy.com/alchemy/documentation/alchemy-web3) API
+* Sign Ethereum transactions using Metamask
+
+For this dApp, we'll be using [React](https://reactjs.org/) as our frontend framework; however, it's important to note that we won't be spending much time breaking down its fundamentals, as we'll mostly be focusing on bringing Web3 functionality to our project.
+
+As a prerequisite, you should have a beginner-level understanding of React—know how components, props, useState/useEffect, and basic function calling works. If you've never heard of any of those terms before, we recommend that you check out this [Intro to React tutorial](https://reactjs.org/tutorial/tutorial.html). For the more visual learners, we highly recommend this awesome Net Ninja [Full Modern React Tutorial](https://www.youtube.com/playlist?list=PL4cUxeGkcC9gZD-Tvwfod2gaISzfRiP9d) video series.
+
+## Step 1: Clone the starter files {#step-1-starter-files}
+
+First, go to the [hello-world-part-four github repository](https://github.com/alchemyplatform/hello-world-part-four-tutorial) to get the starter files for this project. Clone this repository into your local environment.
+
+### Don't know how to clone a repository? Check out [this guide](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/cloning-a-repository) from Github.
+
+When you open this cloned `hello-world-part-four` repository, you'll notice that it contains two folders: `starter-files` and `completed`.
+
+* `starter-files` contains the starter files \(essentially the React UI\) for this project. In this tutorial, **we will be working in this directory**, as you learn how to bring this UI to life by connecting it to your Ethereum wallet and the Hello World smart contract that you published on Etherscan in [Part 3](https://docs.alchemy.com/alchemy/tutorials/hello-world-smart-contract/submitting-your-smart-contract-to-etherscan).
+* `completed` contains the entire completed tutorial and is there for you as a **reference** **if you get stuck.**
+
+Next, open your copy of `starter-files` to your favorite code editor \(at Alchemy, we we're big fans of [VSCode](https://code.visualstudio.com/download)\), and then navigate into your `src` folder.
```

Comment:
Really don't think we need to give people code editor recommendations if they're this far into a web3 programming tutorial